pkg/algorithm/hived_algorithm_test.go

```diff
@@ -509,6 +592,34 @@ func testSuggestedNodes(t *testing.T, h *HivedAlgorithm) {
 	compareSchedulingResult(t, pod, psr)
 }
 
+func testStatefulPreemption(t *testing.T, configFilePath string) {
+	sConfig := api.NewConfig(api.InitRawConfig(&configFilePath))
+	h := NewHivedAlgorithm(sConfig)
+	allocatedPods = []*core.Pod{}
+	var psr internal.PodScheduleResult
+	for i, podName := range casesForStatefulPreemption {
+		pod := allPods[podName]
+		pod.Annotations[api.AnnotationKeyPodSchedulingSpec] = common.ToYaml(pss[pod.UID])
+		psr = h.Schedule(pod, allNodes)
+		compareSchedulingResult(t, pod, psr)
+		if psr.PodBindInfo != nil {
+			allocatedPod := internal.NewBindingPod(pod, psr.PodBindInfo)
+			h.AddAllocatedPod(allocatedPod)
+			allocatedPods = append(allocatedPods, allocatedPod)
+		}
+		if deletedGroups := deletedPreemptorGroups[podName]; deletedGroups != nil {
+			for _, g := range deletedGroups {
```

Comment:
Also test the acquiring and acquired cells are reserved, so non-higher priority AG will not try to schedule to the same cells.

Reply:
see pod30